command/agent_generate_config_test.go

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package command
+
+import (
+	"context"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/vault/api"
+)
+
+// TestConstructTemplates tests the construcTemplates helper function
+func TestConstructTemplates(t *testing.T) {
```

Comment:
after we add the config, should add tests that the generated config should be readable

Reply:
Good call, we definitely should